src/Stratis.Bitcoin.Features.SmartContracts/SmartContractPosPowBlockDefinition.cs

```diff
@@ -67,6 +68,13 @@ public sealed class SmartContractPosPowBlockDefinition : BlockDefinition
             this.stakeChain = stakeChain;
             this.stakeValidator = stakeValidator;
             this.stateRoot = stateRoot;
+
+            // When building smart contract blocks, we will be generating and adding both transactions to the block and txouts to the coinbase. 
+            // At the moment, these generated objects aren't accounted for in the block size and weight accounting. 
+            // This means that if blocks started getting full, this miner could start generating blocks greater than the max consensus block size.
+            // To avoid this without significantly overhauling the BlockDefinition, for now we just lower the block size by a percentage buffer.
+            // If in the future blocks are being built over the size limit and you need an easy fix, just increase the size of this buffer.
```

Comment:
This comment is a repeat of the one in the other file. Is there any way to avoid the duplication ?

Reply:
Sorry Andy, but I think this is more a reflection of the issues with the modularity of the BlockDefinition. Pains me but not going to move it for now. We also don't really know which BlockDefinition we'll end up using.